cpp/src/io/parquet/page_decode.cuh

```diff
@@ -690,6 +690,41 @@ inline __device__ void get_nesting_bounds(int& start_depth,
   }
 }
 
+/**
+ * @brief Updates nesting level offsets for pruned pages of a list column
+ *
+ * This function iterates through the nesting levels of a column and updates the offsets for a list
+ * column. The offset for the current nesting level equals the length of the next nesting level
+ *
+ * @tparam decode_block_size The size of the block used for decoding.
+ * @param[in,out] state Pointer to page state containing column and nesting information.
+ */
+template <int block_size>
+static __device__ void update_list_offsets_for_pruned_pages(page_state_s* state)
+{
+  namespace cg = cooperative_groups;
+
+  int constexpr start_depth    = 0;
+  int const max_depth          = state->col.max_nesting_depth - 1;
+  bool const in_nesting_bounds = (0 >= start_depth and 0 <= max_depth);
```

Comment:
now that start_depth is `constexpr`, it does not seem like we need to check its value
```suggestion
  bool const in_nesting_bounds = max_depth >= 0;
```